thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -143,4 +170,25 @@ export class GameSocket {
     public get onLobbyEvent(): ISimpleEvent<InboundMessage> {
         return this._onLobbyEvent.asEvent();
     }
+
+    public get onGameEvent(): ISimpleEvent<InboundMessage> {
+        return this._onGameEvent.asEvent();
+    }
 }
+
+/**
+ * An error representing a problem with the connection.
+ */
+export class ConnectionError extends Error {
+    /**
+     * Creates a new ConnectionError
+     * @param message A custom error message to display.
+     */
+    constructor(message?: string) {
+        if (message === undefined) {
+            message = "The game connection is missing or is in a broken state.";
+        }
+        super(message);
+        Object.setPrototypeOf(this, new.target.prototype);

```

Comment:
What’s this do?

Reply:
Good question. The answer is I don't really know, but as I understand it, inheriting from the `Error` base class in javascript is weird. The error base class messes with the inheritance tree, so if you don't have that line, the error will appear as a generic error, not a `ConnectionError`. Adding that line after the super call fixes the issue.